pkg/download/bucket/bucket.go

```diff
@@ -0,0 +1,171 @@
+		return result, nil
+	}
+
+	configs := d.convertAllObjects(projectName, response.Objects)
+	result["bucket"] = configs
+	return result, nil
+}
+
+func (d *Downloader) convertAllObjects(projectName string, objects [][]byte) []config.Config {
+	result := make([]config.Config, 0, len(objects))
+	for _, o := range objects {
+		var bucketName struct {
+			BucketName  string `json:"bucketName"`
+			DisplayName string `json:"displayName"`
+		}
+		if err := json.Unmarshal(o, &bucketName); err != nil {
+			return result
+		}
+
+		// exclude builtin bucket names
+		if strings.HasPrefix(bucketName.BucketName, "dt_") || strings.HasPrefix(bucketName.BucketName, "default_") {
+			continue
+		}
+
+		// construct config object with generated config ID
+		configID := idutils.GenerateUUIDFromString(bucketName.BucketName)
+		configCoord := coordinate.Coordinate{
+			Project:  projectName,
+			Type:     "bucket",
+			ConfigId: configID,
+		}
+
+		params := map[string]parameter.Parameter{}
+		{
+			var s string
+			var err error
+
+			s, err = getValueForAttribute(o, "bucketName")
+			if err != nil {
+				log.WithFields(field.Coordinate(coordinate.Coordinate{Project: projectName, Type: "bucket", ConfigId: configID}), field.Error(err)).Warn("Failed to get configuration for %v (%s): %v", configID, "bucket", err)
+				continue
+			}
+			params[config.NameParameter] = &value.ValueParameter{Value: s}
+			o, err = replaceAttributeWith(o, "bucketName", config.NameParameter)
+			if err != nil {
+				log.WithFields(field.Coordinate(coordinate.Coordinate{Project: projectName, Type: "bucket", ConfigId: configID}), field.Error(err)).Warn("Failed to get configuration for %v (%s): %v", configID, "bucket", err)
+				continue
+			}
+
+			s, err = getValueForAttribute(o, "displayName")
+			if err != nil {
+				log.WithFields(field.Coordinate(coordinate.Coordinate{Project: projectName, Type: "bucket", ConfigId: configID}), field.Error(err)).Warn("Failed to get configuration for %v (%s): %v", configID, "bucket", err)
+				continue
+			}
+			if s != "" {
+				params["displayName"] = &value.ValueParameter{Value: s}
+				o, err = replaceAttributeWith(o, "displayName", "displayName")
+				if err != nil {
+					log.WithFields(field.Coordinate(coordinate.Coordinate{Project: projectName, Type: "bucket", ConfigId: configID}), field.Error(err)).Warn("Failed to get configuration for %v (%s): %v", configID, "bucket", err)
+					continue
+				}
+			}
+
+		}
+
+		originObjectID := fmt.Sprintf("%s_%s", configCoord.Project, configCoord.ConfigId)
+
+		c := config.Config{
+			Template:       template.NewDownloadTemplate(configID, configID, string(jsonutils.MarshalIndent(o))),
+			Parameters:     params,
+			Coordinate:     configCoord,
+			Type:           config.BucketType{},
+			OriginObjectId: originObjectID,
+		}
+		result = append(result, c)
+	}
+
+	log.Info("downloaded %d configurations for GRAIL bucket", len(result))
```

Comment:
`len(result)` is always one, right? It should be `len(result["buckets"])`?

I think best you create the results object at the very end and only work with configs until the end